tests/test_gsc_bookmarks.py

```diff
@@ -0,0 +1,165 @@
+                                     "performance_report_query": {site_url: {"image": second_sync_bookmark_date, "web": second_sync_bookmark_date, "video": second_sync_bookmark_date}},
+                                     "performance_report_date": {site_url: {"image": second_sync_bookmark_date, "web": second_sync_bookmark_date, "video": second_sync_bookmark_date}},
+                                     "performance_report_custom": {site_url: {"image": second_sync_bookmark_date, "web": second_sync_bookmark_date, "video": second_sync_bookmark_date}}}}
+
+        menagerie.set_state(conn_id, new_states)
+        LOGGER.info("New Bookmark: {}".format(menagerie.get_state(conn_id)))
+
+        ##########################################################################
+        # Second Sync
+        ##########################################################################
+
+        second_sync_record_count = self.run_and_verify_sync(conn_id)
+        second_sync_records = runner.get_records_from_target_output()
+        second_sync_bookmarks = menagerie.get_state(conn_id)
+
+        ##########################################################################
+        # Test By Stream
+        ##########################################################################
+
+        for stream in expected_streams:
+            with self.subTest(stream=stream):
+
+                # Expected values
+                expected_replication_method = expected_replication_methods[stream]
+
+                # Collect information for assertions from syncs 1 & 2 base on expected values
+                first_sync_count = first_sync_record_count.get(stream, 0)
+                second_sync_count = second_sync_record_count.get(stream, 0)
+                first_sync_messages = [record.get('data') for record in
+                                       first_sync_records.get(
+                                           stream, {}).get('messages', [])
+                                       if record.get('action') == 'upsert']
+
+                second_sync_messages = [record.get('data') for record in
+                                        second_sync_records.get(
+                                            stream, {}).get('messages', [])
+                                        if record.get('action') == 'upsert']
+
+                first_bookmark_value = first_sync_bookmarks.get('bookmarks', {stream: None}).get(stream)
+                second_bookmark_value = second_sync_bookmarks.get('bookmarks', {stream: None}).get(stream)
+
+                if expected_replication_method == self.INCREMENTAL:
+
+                    # Collect information specific to incremental streams from syncs 1 & 2
+                    replication_key = next(iter(expected_replication_keys[stream]))
+
+                    if stream == 'performance_report_page':
+                        replication_key = 'date'
```

Comment:
Can't we keep this replication key of `performance_report_page` in base,py itself? If it is specific due to any reason, then can you add one comment?